ci/tests.sh

```diff
@@ -2,57 +2,22 @@
 
 set -e
 
-VERSION="${TRAVIS_TAG:-0.0.0}"
-VERSION="${VERSION#v}"
-
 readonly ROOT_DIR="$(dirname "$(dirname "${0}")")"
-cd "${ROOT_DIR}"
-
-function test_fluentd_plugin() {
-  local plugin_name="${1}"
-  local version="${2}"
-  # Strip everything after "-" (longest match) to avoid gem prerelease behavior
-  local gem_version="${version%%-*}"
-  local result
-
-  pushd "${plugin_name}" || exit 1
-
-  if [[ -z "${version}" ]] ; then
-    echo "Please provide the version when bundling fluentd plugins"
-    exit 1
-  fi
-
-  echo "Preparing gem ${plugin_name} version ${gem_version} in $(pwd) for testing ..."
-  sed -i.bak "s/0.0.0/${gem_version}/g" ./"${plugin_name}".gemspec
-  rm -f ./"${plugin_name}".gemspec.bak
-
-  echo "Install bundler..."
-  bundle install
-
-  echo "Run unit tests..."
-  bundle exec rake
-
-  readonly result=$?
-  popd || exit 1
-  if [ "${result}" -ne "0" ]; then
-   exit 1
-  fi
-}
-
-function test_fluentd_plugins() {
-  local version="${1}"
-
-  if [[ -z "${version}" ]] ; then
-    echo "Please provide the version when bundling fluentd plugins"
-    exit 1
-  fi
-
-  find . -maxdepth 1 -name 'fluent-plugin-*' -type 'd' -print |
-    while read -r line; do
-      # Run tests in their own context
-      test_fluentd_plugin "$(basename "${line}")" "${version}" || exit 1
-    done
-}
+# shellcheck disable=SC1090
+source "${ROOT_DIR}"/ci/_test_functions.sh
+# shellcheck disable=SC1090
+source "${ROOT_DIR}"/ci/_build_functions.sh
+
+if is_checkout_on_tag; then
+  VERSION="$(git describe --tags --always)"
+  readonly VERSION="${VERSION#v}"
+else
+  # Don't use non-tag 'git describe' output as this will
+  # fail when building fluentd plugins.
+  readonly VERSION="0.0.0"
+fi
```

Comment:
Can we simplify this as well?

Reply:
Done. For the reference: changing this to assign what `git describe --tags` returns will yield in semver versions (e.g. `2.0.0`) for versions that are not a `2.0.0` tag, so `2.0.0-dev.0-136-g205f297` will yields gems with version `2.0.0`